### PR TITLE
Create dedicated page for Styles Development Forum Rules

### PR DESCRIPTION
**Pull request by PayBas**
- [x] Refactor template names and routes
- [x] Create .yml redirects from `/styles/ssp/3.0/` to `/styles/rules-and-policies/submission-policy/3.0/` etc.
- [x] Create rules index page (like extension rules) at `/styles/rules-and-policies/`
- [x] Refactor old `{ style_ssp }` twig vars
- [x] Write the actual forum rules (copy from rules topics)
- [x] Update `/styles/junior-validators/` with info from the [Junior Style Validators](https://www.phpbb.com/community/viewtopic.php?f=74&t=2123500) topic.
- [x] Create .yml redirect from `/styles/contribute/index.php?p=jv` to `/styles/junior-validators/`

**.com (community) to do list:**
- [ ] Create `Validated Styles Support Topics` forum-link under `[3.1.x] Styles Database Releases` to `customise/db/support/style` (just like the 3.0 version)
- [x] Create `[3.1.x] Abandoned Styles` forum under `[3.1.x] Styles in Development`
- [ ] Set `[3.1.x] Abandoned Styles` "List subforum in parent-forum’s legend:" to **no** (consistency)
- [ ] ~~Set `Validated MODs Support Topics` "List subforum in parent-forum’s legend:" to **no** (consistency with extensions forum structure)~~
- [ ] ~~Set `Validated Styles Support Topics` (3.1.x) "List subforum in parent-forum’s legend:" to **no** (consistency with extensions forum structure)~~
- [ ] ~~Set `Validated Styles Support Topics` (3.0.x) "List subforum in parent-forum’s legend:" to **no** (consistency with extensions forum structure)~~

**after merge:** old topics cleanup. All these topics can either be converted to forum-links, or the new page links can be included in the specific forum rules:
- [ ] Replace [The Styles Submission Policy (How to Submit a Style)](https://www.phpbb.com/community/viewtopic.php?f=73&t=988545) topic with forum-link to `/styles/rules-and-policies/submission-policy/3.0/`
- [ ] Replace [How to Post Development Styles](https://www.phpbb.com/community/viewtopic.php?f=476&t=2280051) topic (3.1.x) with forum-link to `/styles/rules-and-policies/development-rules/`
- [ ] Replace [How to Post Development Styles](https://www.phpbb.com/community/viewtopic.php?f=185&t=545785) topic (3.0.x) with forum-link to `/styles/rules-and-policies/development-rules/`
- [ ] Add forum-link in the `[3.1.x] Styles Support & Discussion` to `/styles/support/`
- [ ] Replace [Frequently Asked Questions](https://www.phpbb.com/community/viewtopic.php?f=74&t=988575) topic (3.0.x) forum with forum-link to `/styles/support/`
- [ ] Add forum-link in the `[3.1.x] Styles Support & Discussion` to `/styles/junior-validators/`
- [ ] Replace [Junior Style Validators](https://www.phpbb.com/community/viewtopic.php?f=74&t=2123500) topic (3.0.x) forum with forum-link to `/styles/junior-validators/`

https://www.phpbb.com/community/viewtopic.php?f=25&t=2279851
https://tracker.phpbb.com/browse/WEBSITE-1217
WEBSITE-1217
## the rules (latest version)

untitled-1

### CODE REVIEW

**Comment by PayBas** [comment]
@cyberalien @Raimon @Rahber @michaelcullum  ready for review.

**Comment by PayBas** [comment]
The rules, for those that need visuals.

**Comment by cyberalien** [comment]
1c about security vulnerabilities: does not apply to styles

I suggest to add rule about requiring ether demo or download link.

**Comment by PayBas** [comment]
Ok I'll remove that rule. As for the demo requirements, I listed them with
the topic status/prefixes

**Comment by cyberalien** [comment]
I mean demo or download, not demo or screenshot. All screenshot-only releases so far have been nothing but spam and ended up being abandoned.

**Comment by PayBas** [comment]
Ok, so we want a screenshot AND demo for all stages. And a download for beta/rc/release ?

**Comment by cyberalien** [comment]
No. Everything is optional, but demo or download must be present for all stages. So there wouldn't be any screeshot only releases photoshopped in 1 minute.

**Comment by PayBas** [comment]
"everything is optional" except 2/3 :p 

I tend to find screenshots very useful. Especially when glancing over styles in development. I don't think it's unreasonable to demand a screenshot (just to remind you of which style it is).

**Comment by cyberalien** [comment]
I think you've misread my message. 1/2 is required: demo OR download.

**Comment by PayBas** [comment]
Ok, so I suggest that we make screenshot mandatory for all stages (simply because a style name is not enough recognition).

And we make demo AND/OR download mandatory for all stages?

**Comment by cyberalien** [comment]
That would work.

**Comment by PayBas** [comment]
Updated rules

untitled-1

**Comment by cyberalien** [comment]
Its too strict now. Instead of asking for demo or download, you wrote that demo is mandatory for all stages. 

If download and screenshot are available, there is no point in creating demo. User can see preview on screenshot and download style to properly test it. Since download is required for anything other than dev stage, I suggest this:

dev stage: A screenshot and ether demo or download link must be provided.
other stages: A screenshot and download link must be provided (demo is optional).

**Comment by PayBas** [comment]
Ah yeah, that's right. I'll get back at it XD

**Comment by PayBas** [comment]
Like this?

**Comment by cyberalien** [comment]
Perfect!

**Comment by PayBas** [comment]
Ok, so now we just need @michaelcullum to review the backend stuff, and I think this is ready for merge.

File: src/AppBundle/Resources/views/About/advertise.html.twig [hunk 1]
```diff
@@ -144,7 +144,6 @@
 		<li>All advertisements will be screened by the phpBB.com staff and we reserve the right to reject any campaign that we find unsuitable for this site.</li>
 		<li>We do our best to follow <a href="http://support.google.com/webmasters/bin/topic.py?hl=en&topic=2371375&parent=1724125&ctx=topic">Google's policies</a> regarding advertising and SEO. No advertising campaign should be targeted-toward or based-on pagerank misrepresentation or any other SEO scheme.
 		<li>We reserve the right to discontinue/cancel a campaign at any time. In this case, you will receive a refund for the amount of time remaining on your campaign. For example, if you had a 90 day campaign totaling $300 that was canceled after 66 days, you would receive: [$300 / 90 * (90 - 66) = $80].</li>
-</li>
```
Comment (michaelcullum) [inline src/AppBundle/Resources/views/About/advertise.html.twig]: Is this on purpose?

**Comment by michaelcullum** [comment]
Looks fine to me but I'd rather have confirmation from @Raimon on this first (Unless he's away, in which case @cyberalien's okay will be fine).

File: src/AppBundle/Resources/views/About/advertise.html.twig (hunk 1 above)
Comment (PayBas) [inline src/AppBundle/Resources/views/About/advertise.html.twig]: Yes. It's a rogue tag. Fails HTML validation.

**Comment by cyberalien** [comment]
Yes, lets wait for more feedback.

@Raimon @vinny @Rahber

**Comment by Rahber** [comment]
Looks okay to me.

**Comment by cyberalien** [comment]
Raimon is away, 3 team members are ok with current implementation, so I think this can be merged. It can be edited later anyway.

**Comment by Raimon** [comment]
Sorry guys, my computer is stil broken.
But what i have seen looks fine by me.

Verstuurd vanaf mijn iPhone

**Comment by michaelcukb** [comment]
@PayBas Please hop onto IRC and poke me about this so we can go through and do all of the actions requiring acp access at the same time as merging this.

**Comment by michaelcullum** [comment]
Bump

**Comment by vinny** [comment]
bump

**Comment by michaelcullum** [comment]
@vinny With Arty and Raimon otherwise engaged it's up to you. If you're happy with this, I'll resolve the conflicts and merge this.

**Comment by vinny** [comment]
Go ahead. Arty and Raimon had previously approved.

**Comment by vinny** [comment]
@michaelcullum

**Comment by vinny** [comment]
@michaelcullum bump

**Pull request merged**